src/test/regress/expected/insert_select_null_dist_key.out

```diff
@@ -0,0 +1,606 @@
+HINT:  Try rewriting your queries with 'INSERT INTO ... SELECT' syntax.
+INSERT INTO upsert_test_1 (unique_col, other_col) VALUES (1, 1) ON CONFLICT (unique_col)
+	DO UPDATE SET other_col = random()::int;
+ERROR:  queries that reference a distributed table without a shard key can only reference colocated distributed tables or reference tables
+DETAIL:  functions used in the DO UPDATE SET clause of INSERTs on distributed tables must be marked IMMUTABLE
+INSERT INTO upsert_test_1 (unique_col, other_col) VALUES (1, 1) ON CONFLICT (unique_col)
+	DO UPDATE SET other_col = 5 WHERE upsert_test_1.other_col = random()::int;
+ERROR:  queries that reference a distributed table without a shard key can only reference colocated distributed tables or reference tables
+DETAIL:  functions used in the WHERE clause of the ON CONFLICT clause of INSERTs on distributed tables must be marked IMMUTABLE
+INSERT INTO upsert_test_1 (unique_col, other_col) VALUES (1, 1) ON CONFLICT (unique_col) WHERE unique_col = random()::int
+	DO UPDATE SET other_col = 5;
+ERROR:  queries that reference a distributed table without a shard key can only reference colocated distributed tables or reference tables
+DETAIL:  functions used in the WHERE clause of the ON CONFLICT clause of INSERTs on distributed tables must be marked IMMUTABLE
+-- test upsert with INSERT .. SELECT queries
+INSERT INTO upsert_test_1 (unique_col, other_col) SELECT unique_col, other_col FROM upsert_test_1 ON CONFLICT (unique_col)
+    DO UPDATE SET other_col = upsert_test_1.other_col + 1;
+DEBUG:  distributed statement: INSERT INTO insert_select_null_dist_key.upsert_test_1_1820106 AS citus_table_alias (unique_col, other_col) SELECT upsert_test_1.unique_col, upsert_test_1.other_col FROM insert_select_null_dist_key.upsert_test_1_1820106 upsert_test_1 ON CONFLICT(unique_col) DO UPDATE SET other_col = (citus_table_alias.other_col OPERATOR(pg_catalog.+) 1)
+-- TODO: this looks interesting
+INSERT INTO upsert_test_1 (unique_col, other_col) SELECT unique_col, other_col FROM upsert_test_1 ON CONFLICT (unique_col)
+	DO UPDATE SET other_col = (SELECT count(*) from upsert_test_1);
+DEBUG:  distributed statement: INSERT INTO insert_select_null_dist_key.upsert_test_1_1820106 AS citus_table_alias (unique_col, other_col) SELECT upsert_test_1.unique_col, upsert_test_1.other_col FROM insert_select_null_dist_key.upsert_test_1_1820106 upsert_test_1 ON CONFLICT(unique_col) DO UPDATE SET other_col = (SELECT count(*) AS count FROM insert_select_null_dist_key.upsert_test_1)
+ERROR:  cannot execute a distributed query from a query on a shard
+DETAIL:  Executing a distributed query in a function call that may be pushed to a remote node can lead to incorrect results.
+HINT:  Avoid nesting of distributed queries or use alter user current_user set citus.allow_nested_distributed_execution to on to allow it with possible incorrectness.
+CONTEXT:  while executing command on localhost:xxxxx
```

Comment:
This doesn't look correct, looking into that.

Reply:
Turns out that this is what we have on main too: https://github.com/citusdata/citus/issues/6826.